client/src/components/ObjectStore/SelectObjectStore.vue

```diff
@@ -63,57 +45,72 @@ const error = computed(() => {
     return props.parentError || loadErrorMessage.value;
 });
 
-async function handleSubmit(preferredObjectStore: ConcreteObjectStoreModel | null) {
-    const id: string | null = (preferredObjectStore ? preferredObjectStore.object_store_id : null) as string | null;
+async function handleSubmit(preferredObjectStore: UserConcreteObjectStoreModel) {
+    let id = preferredObjectStore?.object_store_id ?? null;
     const isPrivate: boolean = preferredObjectStore ? preferredObjectStore.private : false;
+
+    if (id === "__null__") {
+        id = null;
+    }
+
     emit("onSubmit", id, isPrivate);
 }
+
+const defaultObjectStore: UserConcreteObjectStoreModel = {
+    object_store_id: "__null__",
+    name: props.defaultOptionTitle,
+    description: props.defaultOptionDescription,
+    badges: [],
+    private: false,
+    quota: { enabled: true, source: null },
+    active: false,
+    hidden: false,
+    purged: false,
+    secrets: [],
+    template_id: "",
+    template_version: 0,
+    type: "disk",
+    uuid: "",
+    variables: null,
+};
 </script>
 
 <template>
     <div>
-        <LoadingSpan v-if="isLoading" :message="loadingObjectStoreInfoMessage" />
+        <LoadingSpan v-if="isLoading" message="Loading storage location information" />
         <div v-else>
-            <b-alert v-if="error" variant="danger" class="object-store-selection-error" show>
+            <span v-localize>
+                Select a preferred storage location for new datasets. Depending on the job and workflow execution
```

Comment:
Can you make the extra text dependent on the option again? I think adding that option in was a solid usability enhancement for main.